lib/lrama/command.rb

```diff
@@ -5,65 +5,104 @@ class Command
     LRAMA_LIB = File.realpath(File.join(File.dirname(__FILE__)))
     STDLIB_FILE_PATH = File.join(LRAMA_LIB, 'grammar', 'stdlib.y')
 
-    def run(argv)
-      parser_options(argv)
+    def initialize(argv)
+      @logger = Lrama::Logger.new
+      @options = OptionParser.parse(argv)
+      @tracer = Tracer.new(@logger, **@options.trace_opts)
+      @reporter = Reporter.new(**@options.report_opts)
+      @warnings = Warnings.new(@logger, @options.warnings)
+    rescue => e
+      message = e.message
```

Comment:
Can we use `#format_error_message`?

Reply:
Yes, thank you. Change: ead9c9481b353365abcc3cff8b36780c6bc70e47